common/src/main/java/org/apache/gravitino/json/JsonUtils.java

```diff
@@ -479,6 +502,85 @@ private static void writeFunctionArg(FunctionArg arg, JsonGenerator gen) throws
     gen.writeEndObject();
   }
 
+  private static void writeExpression(Expression expression, JsonGenerator gen) throws IOException {
```

Comment:
I guess the implementation is quite similar to `xxxDTO` serde, maybe we can reuse the code. Otherwise, we have to change twice.

Another solution is that when we store these into the database, we can convert the `xxxImpl` to `xxxDTO`, and then using the existing serde method to do the serialization and deserialization. With this, we don't have to implement the serde method for these `xxxImpl`.

What do you think @yuqi1129 @mchades ?

Reply:
In the first edition, I have just converted Impl to DTO and reused the DTO serializer and deserializer to handle it. However, since DTO and Impl play different roles, I chose to create new serializers finally. Theoretically, both are okay.

Reply:
Let's do the second one if there's no objection, cc @mchades .